### PR TITLE
fix: Log print duplicate key val

### PR DESCRIPTION
**Pull request by demoManito**
#### Description (what this PR does / why we need it):
- [x] (De duplication of duplicate log key values)

#### Which issue(s) this PR fixes (resolves / be part of):
- [x] fix issues: #2347

#### Other special notes for the reviewers:

I don't think it is necessary to de duplicate data by using `func With(l Logger, kv ...interface{})`, which may cause the output to be inconsistent with expectations.
`func WithUniq(l Logger, key string, val interface{})` can be used to ensure that the key and val are not duplicated. Users can select this function according to their needs

我认为没有必要在使用 `With` 时消除重复数据，这可能会导致输出与预期不一致。
`WithUniq` 可用于确保 KEY 不重复，并且保证 KEY 对应的 VAL 是最新的，用户可以根据需要自行选择此功能。

### CODE REVIEW

File: log/log.go [hunk 1]
```diff
@@ -27,12 +27,33 @@ func (c *logger) Log(level Level, keyvals ...interface{}) error {
...
 		return err
 	}
 	return nil
 }
 
+// uniqKeys Under the same key, the Val corresponding to the last key shall prevail
+func uniqKeys(kvs []interface{}) []interface{} {
+	uniqMap := make(map[interface{}]interface{})
+	keys := make([]interface{}, 0, len(kvs))
+	for i := 0; i < len(kvs); i++ {
```
Comment (icylight) [inline log/log.go:40]: for i:=0;i<len(kvs); i = i+2

File: log/log.go [hunk 2]
```diff
@@ -27,12 +27,33 @@ func (c *logger) Log(level Level, keyvals ...interface{}) error {
...
+		if i&1 == 1 { // only the key is checked for repeatability
+			continue
+		}
```
Comment (icylight) [inline log/log.go:43]: 这里不用要

**Comment by shenqidebaozi** [comment]
#2350 Repeat?

**Comment by demoManito** [comment]
> [第2350章](https://github.com/go-kratos/kratos/pull/2350)重复？

yes, I read # 2350 It may not be right to handle it at the same time

**Comment by hrygo** [comment]
> #2350 Repeat?

我的实现方式保持了原有顺序，相同的key，保留后来者。我个人认为这两点比较重要。

**Comment by tonybase** [comment]
这个会导致log性能变差，不太建议再做一次对象

**Pull request closed without merging**